frontend/packages/kubevirt-plugin/integration-tests/tests/vm.environment.scenario.ts

```diff
@@ -0,0 +1,133 @@
+const vmName = 'vm-example';
+const configmapName = 'configmap-mock';
+const secretName = 'secret-mock';
+const serviceAccountName = 'service-account-mock';
+
+describe('Test VM enviromnet tab', () => {
+  const secret = getSecret(testName, secretName);
+  const configmap = getConfigMap(testName, configmapName);
+  const serviceAccount = getServiceAccount(testName, serviceAccountName);
+  let vm: VirtualMachine;
+
+  beforeAll(async () => {
+    createResource(secret);
+    createResource(configmap);
+    createResource(serviceAccount);
+    vm = await createExampleVMViaYAML(true);
+  });
+
+  afterAll(() => {
+    deleteResource(secret);
+    deleteResource(configmap);
+    deleteResource(serviceAccount);
+  });
+
+  beforeEach(async () => {
+    await browser.get(`${appHost}/k8s/ns/${testName}/virtualmachines/${vmName}/environment`);
+    await isLoaded();
+  });
+
+  it('Verify vm object was created', () => {
+    expect(!!vm).toBeTruthy();
+  });
+
+  it('Add configmap, secret and service account', async () => {
+    await vmEnv.addSource(configmapName);
+    await vmEnv.addSource(secretName);
+
+    // Add Service Account
+    await browser.wait(until.elementToBeClickable(vmEnv.addVariableButton), PAGE_LOAD_TIMEOUT_SECS);
+    await click(element(by.buttonText('Add All From Config Map or Secret')));
+    await addVariableFrom(serviceAccountName, null, true);
+    await browser.wait(until.presenceOf(vmEnv.successAlert));
+    expect(vmEnv.successAlert.isDisplayed()).toEqual(true);
+  });
+
+  it('Verify all the sources are present at the disks tab', async () => {
+    const disks = await vm.getAttachedDisks();
+    expect(!!disks.find((d) => d.name.includes(configmapName))).toBeTruthy();
+    expect(!!disks.find((d) => d.name.includes(secretName))).toBeTruthy();
+    expect(!!disks.find((d) => d.name.includes(serviceAccountName))).toBeTruthy();
+  });
+
+  it(
+    'Verify all sources are readable inside the VM',
+    async () => {
+      await vm.action(VM_ACTION.Start);
+      const out = execSync(`expect ${expecScriptPath} ${vmName} ${testName}`).toString();
+      const isFailedTest = out.split('\n').find((line) => line === 'FAILED');
+      expect(!isFailedTest).toBeTruthy();
+    },
+    VM_BOOTUP_TIMEOUT_SECS * 2, // VM boot time + test sources time
+  );
+
+  it('Error when resource has no serial', async () => {
+    await vmEnv.serialField.get(1).clear();
+    await vmEnv.serialField.get(1).sendKeys('i', Key.BACK_SPACE); // workaround: for some reason clear() is not enough
+    await browser.wait(until.elementToBeClickable(vmEnv.saveBtn), PAGE_LOAD_TIMEOUT_SECS);
+    await vmEnv.saveBtn.click();
+    await browser.wait(until.presenceOf(vmEnv.errorAlert), PAGE_LOAD_TIMEOUT_SECS);
+    const errorText = await vmEnv.errorAlert.getText();
+    expect(vmEnv.errorAlert.isDisplayed()).toEqual(true);
+    expect(errorText.includes(vmEnv.noSerialError)).toEqual(true);
```

Comment:
Optional:
`expect(errorText.includes(vmEnv.noSerialError)).toEqual(true);`
can be written as 
`expect(errorText).toContain(vmEnv.noSerialError);`
which is slightly nicer to read.